packages/launcher/lib/linux/index.ts

```diff
@@ -3,8 +3,33 @@ import type { FoundBrowser, Browser, PathData } from '../types'
 import { notInstalledErr } from '../errors'
 import { utils } from '../utils'
 import os from 'os'
+import { promises as fs } from 'fs'
 import path from 'path'
 import Bluebird from 'bluebird'
+import which from 'which'
+
+async function isFirefoxSnap (binary: string): Promise<boolean> {
+  try {
+    return await Bluebird.resolve((async () => {
+      const binaryPath = await which(binary)
+
+      // read the first 16kb, don't read the entire file into memory in case it is a binary
+      const fd = await fs.open(binaryPath, 'r')
+      // @ts-ignore - needs @types/node at least 16
+      // https://github.com/cypress-io/cypress/issues/21329
+      const { buffer, bytesRead } = await fd.read<Buffer>({ length: 16384 })
+
+      await fd.close()
+
+      return buffer.slice(0, bytesRead).toString('utf8').includes('exec /snap/bin/firefox')
```

Comment:
Today I learned about reading only the opening bytes of a file! This seemed odd (particularly the slice), so I poked around the docs and command line, and lo, you have done it correctly. :)